contracts/strategy/LiquidationStrategy.sol

```diff
@@ -230,180 +319,159 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
-         * |                  | dexAmountOut > amountNeededToPayDebt | and deposit from LiquidationStrategy |
-         * |                  | and deposit FXD from WXDCSwap        |                                      |
-         * +------------------+--------------------------------------+--------------------------------------+
-         */
-
-        if (allowLoss == false) {
-            // Condition #1 if there is no loss, sell on DEX
-            if (dexAmountOut >= amountNeededToPayDebt) {
-                uint256 fathomStablecoinReceived = _sellCollateral(
-                    _vars.tokenAdapter.collateralToken(),
-                    path,
-                    _vars.router,
-                    _collateralAmountToLiquidate,
-                    dexAmountOut
-                );
-                if (fathomStablecoinReceived < amountNeededToPayDebt) {
-                    require(fathomStablecoin.balanceOf(address(this)) >= amountNeededToPayDebt, "flashLendingCall: not enough to repay debt");
-                }
-                _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
-                emit LogFlashLiquidationSuccess(
-                    _vars.liquidatorAddress,
-                    _debtValueToRepay,
-                    _collateralAmountToLiquidate,
-                    dexAmountOut,
-                    fathomStablecoinReceived,
-                    path
-                );
-            } else {
-                // Condition #2 if there is loss, don't sell on DEX
-                require(fathomStablecoin.balanceOf(address(this)) >= amountNeededToPayDebt, "flashLendingCall: not enough to repay debt");
-                _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
-                idleWXDC.WXDCAmount += retrievedCollateralAmount;
-                idleWXDC.amountNeededToPayDebt += amountNeededToPayDebt;
-                idleWXDC.averagePriceOfWXDC = idleWXDC.amountNeededToPayDebt.mul(WAD).div(idleWXDC.WXDCAmount);
-                emit LogFlashLiquidationSuccess(
-                    _vars.liquidatorAddress,
-                    _debtValueToRepay,
-                    _collateralAmountToLiquidate,
-                    dexAmountOut,
-                    amountNeededToPayDebt,
-                    path
-                );
+        address _collateralToken = _vars.tokenAdapter.collateralToken();
+        // Retrieve collateral token from CollateralTokenAdapter
+        uint256 retrievedCollateralAmount = _retrieveCollateral(_collateralToken, _vars.tokenAdapter, _collateralAmountToLiquidate);
+        // +1 to compensate for precision loss with division
+        uint256 amountNeededToPayDebt = _debtValueToRepay.div(RAY) + 1;
+        uint256 fathomStablecoinReceivedV2;
+        uint256 fathomStablecoinReceivedV3;
+        if (uniswapV3Info[_vars.routerV3].permit2 == address(0) && _vars.routerV2 != address(0)) {
```

Comment:
we still have redundant sload here we can just do this :
`if (_vars.routerV3== address(0) && _vars.routerV2 != address(0)) {`